src/store/utils.js

```diff
@@ -89,3 +91,189 @@ export const saveCurrentWeatherToLocalStorage = currentWeather => {
+
+const getHourlyForecastAPIUrl = ({ latitude, longitude, APIKey, language = 'uk' }) => {
+  return `https://api.weatherbit.io/v2.0/forecast/hourly?key=${APIKey}&lang=${language}&lat=${latitude}&lon=${longitude}`;
+};
+
+export const getHourlyForecastForCoordinates = async ({ latitude, longitude }) => {
+  const hourlyForecastAPIUrl = getHourlyForecastAPIUrl({
+    latitude,
+    longitude,
+    APIKey: process.env.VUE_APP_WEATHERBIT_KEY,
+  });
+  const response = await fetch(hourlyForecastAPIUrl);
+  if (!response.ok) {
+    throw new Error(`HTTP error, status = ${response.status}`);
+  }
+  const hourlyForecastData = await response.json();
+  return hourlyForecastData;
+};
+
+const getWindDirection = deg => {
+  const rumb = Math.ceil((deg - 11.25) / 22.5);
+  const windDirectionTranslator = {
+    0: 'Пн',
+    1: 'Пн-Пн-С',
+    2: 'Пн-С',
+    3: 'Пн-С-С',
+    4: 'С',
+    5: 'Пд-С-С',
+    6: 'Пд-С',
+    7: 'Пд-Пд-С',
+    8: 'Пд',
+    9: 'Пд-Пд-З',
+    10: 'Пд-З',
+    11: 'Пд-З-З',
+    12: 'З',
+    13: 'Пн-З-З',
+    14: 'Пн-З',
+    15: 'Пн-Пн-З',
+    16: 'Пн',
+  };
+
+  return windDirectionTranslator[rumb];
+};
+
+const mapWeatherbitIconCodeToStandard = weatherbitIconCode => {
+  const iconsMapping = {
+    t01d: '15',
+    t02d: '15',
+    t03d: '15',
+    t04d: '15',
+    t05d: '15',
+    d01d: '18',
+    d02d: '18',
+    d03d: '18',
+    r01d: '18',
+    r02d: '18',
+    r03d: '18',
+    r04d: '18',
+    r05d: '12',
+    r06d: '12',
+    f01d: '19',
+    s01d: '22',
+    s02d: '22',
+    s03d: '22',
+    s04d: '29',
+    s05d: '25',
+    s06d: '19',
+    a01d: '11',
+    a02d: '11',
+    a03d: '5',
+    a04d: '11',
+    a05d: '11',
+    a06d: '11',
+    c01d: '1',
+    c02d: '2',
+    c03d: '4',
+    c04d: '7',
+    t01n: '15',
+    t02n: '15',
+    t03n: '15',
+    t04n: '15',
+    t05n: '15',
+    d01n: '18',
+    d02n: '18',
+    d03n: '18',
+    r01n: '18',
+    r02n: '18',
+    r03n: '18',
+    r04n: '18',
+    r05n: '12',
+    r06n: '12',
+    f01n: '19',
+    s01n: '22',
+    s02n: '22',
+    s03n: '22',
+    s04n: '29',
+    s05n: '25',
+    s06n: '19',
+    a01n: '11',
+    a02n: '11',
+    a03n: '5',
+    a04n: '11',
+    a05n: '11',
+    a06n: '11',
+    c01n: '33',
+    c02n: '34',
+    c03n: '36',
+    c04n: '7',
+  };
+  return iconsMapping[weatherbitIconCode] || weatherbitIconCode;
+};
+
+const formatTemperature = temperature => {
+  return temperature > 0 ? `+ ${Math.round(temperature)}` : Math.round(temperature);
+};
+
+const getDisplayStringByDate = date => {
+  if (isToday(date)) return 'Сьогодні';
+  if (isTomorrow(date)) return 'Завтра';
+  return date.toLocaleDateString('uk-UA', {
+    weekday: 'long',
+  });
+};
+
+const getAllDatesForHeader = jsonDataArray => {
+  const datesArrayAsDate = jsonDataArray.map(arrayItem =>
+    startOfDay(new Date(arrayItem.timestamp_local)),
+  );
+
+  const getUniqueItems = items => [...new Set(items)];
+
+  const datesArrayInms = datesArrayAsDate.map(arrayItem => {
```

Comment:
What does `Inms` stands for?

Reply:
Why not `InMs` then? 🤔

Reply:
Because InMs refers to Megaseconds :grinning:

Reply:
It's generally not a good idea to add types to variable name since when you change type you'll have to change variable name